cryptoprotocol/src/main/java/de/upb/crypto/incentive/cryptoprotocol/model/Token.java

```diff
@@ -1,4 +1,144 @@
 package de.upb.crypto.incentive.cryptoprotocol.model;
 
-public class Token {
+
+import de.upb.crypto.craco.common.GroupElementPlainText;
+import de.upb.crypto.craco.sig.interfaces.VerificationKey;
+import de.upb.crypto.craco.sig.sps.eq.SPSEQSignature;
+import de.upb.crypto.craco.sig.sps.eq.SPSEQSignatureScheme;
+import de.upb.crypto.incentive.cryptoprotocol.exceptions.PedersenException;
+import de.upb.crypto.incentive.cryptoprotocol.exceptions.SPSEQException;
+import de.upb.crypto.math.interfaces.structures.GroupElement;
+import de.upb.crypto.math.serialization.Representable;
+import de.upb.crypto.math.serialization.Representation;
+import de.upb.crypto.math.serialization.annotations.v2.ReprUtil;
+import de.upb.crypto.math.serialization.annotations.v2.Represented;
+import de.upb.crypto.math.structures.zn.Zn;
+import de.upb.crypto.math.structures.zn.Zn.ZnElement;
+
+
+/**
+ * class representing a token from a mathematical point of view (meaning: as a bunch of group elements and exponents modulo p from which you can compute a Pedersen commitment)
+ * serialized representation of token does not contain secret exponents and token plain text (latter can be computed trivially)
+ */
+public class Token implements Representable
+{
+    @Represented
+    private Zn myRemainderClassRing; // the remainder class ring the token's exponents live in
```

Comment:
No need to store this here. It's already stored in the `pp`. Generally, it's somewhat dangerous to store stuff multiple times. If you need access to this, store `pp` here, instead. 

Same for the `commitmentBases`, `userSecretKey`, etc. (they're all stored somewhere else). Why are those in here?

Reply:
I started off with the Token class and implemented the Setup class later on.
Thanks for the comment, I will rewrite it :)